src/components/LmsConfigurations/DegreedIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,356 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+import { handleErrors, validateLmsConfigForm } from './common';
+
+export const REQUIRED_DEGREED_CONFIG_FIELDS = [
+  'degreedBaseUrl',
+  'degreedUserId',
+  'degreedUserPassword',
+  'secret',
+  'degreedCompanyId',
+  'key',
+];
+
+function configFormReducer(state, action) {
+  switch (action.type) {
+    case 'PENDING': {
```

Comment:
Question mostly for my understanding: if you don't update the invalidFields on these other states they retain previous value, correct? Is that an issue?